scrapy/core/scraper.py

```diff
@@ -147,6 +147,14 @@ def _scrape2(self, request_result, request, spider):
 
     def call_spider(self, result, request, spider):
         result.request = request
+        # StopDownload exceptions: make the partial response an attribute of the failure
+        if (
+            isinstance(result, Failure)
+            and isinstance(result.value, StopDownload)
+            and hasattr(result.value, "response")
+        ):
+            result.response = result.value.response
```

Comment:
Actually I wonder if we should just keep failure.value.response as-is. This seems to be consistent with https://docs.scrapy.org/en/latest/topics/request-response.html#using-errbacks-to-catch-exceptions-in-request-processing example.

Reply:
Good catch. It would make this change simpler too.

Reply:
Updated @kmike :tada: